pkg/cluster/kubeclient/roundtrip.go

```diff
@@ -156,19 +194,94 @@ func NewKubeclient(log *logrus.Entry, config *v1.Config, disableKeepAlives bool)
 		}
 	}
 
+	return newKubeclientFromRestConfig(log, restconfig, disableKeepAlives, privateEndpointIP, testConfig)
+}
+
+// EnablePrivateEndpointRoundTripper will override dialers to call
+// API server via PrivateEndpoint while keeping certificate trust
+func (u *Kubeclientset) EnablePrivateEndpointRoundTripper(cs *api.OpenShiftManagedCluster) error {
+	if cs.Properties.NetworkProfile.PrivateEndpoint == nil {
+		return fmt.Errorf("enablePrivateEndpointRoundTripper failed. privateEndpoint is not set")
+	}
+	u.Log.Debugf("override kubeClient roundtripper with PrivateEndpoint dialer")
+	restconfig := u.restconfig
+	tlsConfig, err := restclient.TLSConfigFor(restconfig)
+	if err != nil {
+		return err
+	}
+
+	restconfig.WrapTransport = func(rt http.RoundTripper) http.RoundTripper {
+		var rtNew *http.Transport
+
+		// This is development code. This should never ever run in production
+		if u.testConfig.RunningUnderTest {
```

Comment:
This stinks little bit. But due `DialTLS` this is problematic. Any better way?

Reply:
can we talk about it on Monday - I don't fully understand the issue.

Reply:
sure